.travis/install.sh

```diff
@@ -3,60 +3,70 @@
 set -e
 set -x
 
+if [[ "${MACAPP_ENV}" == "system" ]]; then
+    # If we are testing the mac app environment, it's a bit of a special case;
+    # the only setup we need to do is to ensure that homebrew python is the
+    # default.
+
+    brew update;
+    brew install python;
+    exit 0;
+fi;
+
 if [[ "$(uname -s)" == 'Darwin' ]]; then
     DARWIN=true
 else
     DARWIN=false
 fi
 
-if [[ "$DARWIN" = true ]]; then
+# Do we need to use a Python from pyenv, or will the system one suffice?
+PYENV_PYTHON="";
 
-    brew update
+if [[ "${TOXENV}" == "pypy" ]]; then
+    PYENV_PYTHON="pypy-4.0.1";
+fi;
 
-    if which pyenv > /dev/null; then
-        eval "$(pyenv init -)"
-    fi
+# If we need a python not available in Travis's environment, install pyenv in a
+# way appropriate to the platform.
 
-    case "${TOXENV}" in
-        py27)
-            brew upgrade pyenv
-            pyenv install 2.7.9
-            pyenv global 2.7.9
-            ;;
-        pypy)
-            brew upgrade pyenv
-            pyenv install pypy-2.6.1
-            pyenv global pypy-2.6.1
-            ;;
-    esac
-    pyenv rehash
-    pip install --user virtualenv
-else
-    # temporary pyenv installation to get pypy-2.6 before container infra upgrade
-    if [[ "${TOXENV}" == "pypy" ]]; then
-        git clone https://github.com/yyuu/pyenv.git ~/.pyenv
-        PYENV_ROOT="$HOME/.pyenv"
-        PATH="$PYENV_ROOT/bin:$PATH"
-        eval "$(pyenv init -)"
-        pyenv install pypy-2.6.1
-        pyenv global pypy-2.6.1
+if [[ -n "${PYENV_PYTHON}" ]]; then
+    if [[ "$DARWIN" = true ]]; then
```

Comment:
this weirdness was all here before the change, it's somewhat less inconsistent after the change :).  The mac app env is a particularly weird edge case that has to work around some bugs (you can see the app-building shell script for an explanation).

I think the `"true"` thing is maybe a problem though, I'll look at that.